openhtf/io/frontend/src/app/utils.ts

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 Google Inc.All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import {Pipe, PipeTransform} from 'angular2/core';
+
+
+/** A base class for map-based piping. Subclasses need to override 'map'.**/
+export class MapPipe implements PipeTransform {
+  map = {}
+
+  /**
+   * Return the mapped string for the given input string.
+   */
+  transform(input: string): string {
+    if (this.map[input]) {
+      return this.map[input];
+    }
+    return input;
+  }
+}
+
+
+/** A Pipe for condensing status codes down to a simpler set.**/
+@Pipe({ name: 'simplifyStatus' })
+export class SimplifyStatus extends MapPipe {
+  map = {
+    'EXECUTING': 'RUNNING',
+    'START_WAIT': 'WAITING',
+    'STOP_WAIT': 'WAITING',
+  };
+}
+
+
+/** A Pipe for turning status codes into materialize-css color classes.**/
+@Pipe({name: 'statusToColor'})
+export class StatusToColor extends MapPipe {
+  map = {
+    'RUNNING': 'blue lighten-1 white-text',
+    'ONLINE':  'blue lighten-1 white-text',
+    'PASS':    'green lighten-1 white-text',
+    'FAIL':    'red lighten-1 white-text',
+    'ERROR':   'yellow darken-3 white-text',
+    'PENDING': 'blue-grey lighten-3 white-text',
+    'WAITING': 'blue-grey lighten-3 white-text',
+    'OFFLINE': 'blue-grey lighten-3 white-text',
+  };
+}
+
+
+/** A Pipe for darkening status colors. **/
+@Pipe({name: 'colorToDark'})
+export class ColorToDark extends MapPipe {
```

Comment:
this is only used after a statusToColor, same with ColorToSuperDark. I'd just make them statusToColor, statusToDark, statusToSuperDark. That'll eliminate an extra pipe and having the non-dark color repeated 3 times.

It may instead make sense to create CSS classes with the name of the status, such as status-running, status-online, etc with the color and white-text, then status-light, status-dark and status-super-dark having the lightens and darkens. Then there would only be one pipe 'statusToCss'